From: Petr Tesarik <ptesarik@suse.com>
Subject: Add product identifying information to VMCOREINFO
Patch-mainline: Never, SUSE-specific
References: bsc#1123015

Since SUSE kernels sometimes backport patches that change the structures
required to save a dump file using makedumpfile, it is necessary to recognize
such a patched kernel from a vanilla kernel in the kdump environment. The
VMCOREINFO note is perfect for this purpose.

Signed-off-by: Petr Tesarik <ptesarik@suse.com>
---
 kernel/crash_core.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/kernel/crash_core.c
+++ b/kernel/crash_core.c
@@ -9,6 +9,7 @@
 #include <linux/vmalloc.h>
 #include <linux/memblock.h>
 #include <linux/swiotlb.h>
+#include <linux/suse_version.h>
 
 #ifdef CONFIG_KEXEC_CORE
 #include <asm/kexec.h>
@@ -610,6 +611,7 @@ static int __init crash_save_vmcoreinfo_
 	}
 
 	VMCOREINFO_OSRELEASE(init_uts_ns.name.release);
+	VMCOREINFO_NUMBER(SUSE_PRODUCT_CODE);
 	add_build_id_vmcoreinfo();
 	VMCOREINFO_PAGESIZE(PAGE_SIZE);
 
